subsys/shell/backends/shell_telnet.c

```diff
@@ -13,7 +13,7 @@
 #include <net/net_pkt.h>
 #include <shell/shell_telnet.h>
 
-#include "shell_telnet_protocol.h"
+#include "../shell_telnet_protocol.h"
```

Comment:
I think you should move corresponding private header as well, it's only used by telnet backend.